src/boot/measure.c

```diff
@@ -759,13 +789,55 @@ static int verb_sign(int argc, char *argv[], void *userdata) {
         /* When signing we only support JSON output */
         arg_json_format_flags &= ~JSON_FORMAT_OFF;
 
-        privkeyf = fopen(arg_private_key, "re");
-        if (!privkeyf)
-                return log_error_errno(errno, "Failed to open private key file '%s': %m", arg_private_key);
+        /* This must be done before openssl_load_key_from_token() otherwise it will get stuck */
```

Comment:
hmm?

Reply:
Yes, it is bizarre, but with my yubikey the actual signing gets stuck unless the cert is loaded before the engine is loaded, it took a while to debug so I left a comment given how weird it is

Reply:
I don't think I have the strength in me to go debug the openssl+pkcs11 innards